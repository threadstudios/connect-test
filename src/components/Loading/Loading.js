import Inferno from 'inferno';

export default ({className}) => {
    const classNames = ["loading", className].join(" ");
    return (
        <div className={classNames}>
            <svg width="120" className="loader-svg" height="120" viewbox="0 0 1 1">
                <circle strokeWidth="9" fill="transparent" stroke="#EFEFEF" cx="60" cy="60" r="30"></circle>
                <circle className="loader-ring" strokeWidth="9" fill="transparent" stroke="#9e1b32" cx="60" cy="60" r="30"></circle>
            </svg>
        </div>
    );
}