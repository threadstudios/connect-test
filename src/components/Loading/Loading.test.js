import Inferno from 'inferno';
import { renderToSnapshot, renderIntoDocument, findRenderedDOMElementWithClass } from 'inferno-test-utils';
import Loading from './Loading';

describe("A Loading Component", () => {

    it("should render without an error", () => {
        expect(renderToSnapshot(<Loading />)).toMatchSnapshot();
    })

    it("should take a passed class", () => {
        const Loader = renderIntoDocument(<Loading className="foo" />);
        const result = findRenderedDOMElementWithClass(Loader, "loading");
        expect(result.className).toBe("loading foo");
    })

})