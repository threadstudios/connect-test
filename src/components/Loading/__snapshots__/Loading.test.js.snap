// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Loading Component should render without an error 1`] = `
<div>
  <svg
    height="120"
    viewbox="0 0 1 1"
    width="120"
  >
    <circle
      cx="60"
      cy="60"
      fill="transparent"
      r="30"
      stroke="#EFEFEF"
      stroke-width="9"
    />
    <circle
      cx="60"
      cy="60"
      fill="transparent"
      r="30"
      stroke="#9e1b32"
      stroke-width="9"
    />
  </svg>
</div>
`;
